### PR TITLE
ClimateMapped How It Works

### LINKED ISSUE #916: ClimateMapped How it Works Block
Create How it Works block for ClimateMapped

Request Link: [Feature 1](https://docs.google.com/document/d/1TUrdCRnbj6TSqiuwITKUpnDUEi_ktm_3PmgsaD0y2bw/edit#heading=h.7grsuhlop4fo)

> Preview
Image

> Block Fields
Image

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description

This PR implements How It Works Component.

Fixes #916

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots
image
image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/climatemappedafrica/src/components/HowItWorks/index.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import { Box, Button, Grid, Typography, useMediaQuery } from "@mui/material";
```
Comment (koechkevin) [inline apps/climatemappedafrica/src/components/HowItWorks/index.js:1]: Ensure the components follow the standard Folder/{index.js},{Folder.js},{Folder.test.js @kelvinkipruto

**Review by koechkevin — APPROVED** [review]

**Review by koechkevin — COMMENTED** [review]
👍

**Comment by kelvinkipruto** [comment]
@koechkevin @kilemensi This is ready for review

File: apps/climatemappedafrica/src/payload/blocks/HowItWorks.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          required: true,
+        },
+        {
+          name: "type",
+          type: "text",
+          required: true,
+          defaultValue: "video/youtube",
+          admin: {
+            description: "The type of video. e.g. video/mp4, video/youtube",
+          },
```
[the remark this answers is not in the document]
Comment (kelvinkipruto) [reply apps/climatemappedafrica/src/payload/blocks/HowItWorks.js:35]: Not sure. I think we can default to youtube for now.

**Comment by koechkevin** [comment]
This should go as well @kelvinkipruto

**Review by koechkevin — APPROVED** [review]

File: apps/climatemappedafrica/src/payload/blocks/HowItWorks.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kilemensi) [reply apps/climatemappedafrica/src/payload/blocks/HowItWorks.js]: My main point was if it's youtube, does type change? ... and would users know which type to enter if type does indeed change on youtube?

File: apps/climatemappedafrica/src/components/HowItWorks/index.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            left: 0,
+            top: theme.typography.pxToRem(42),
+            width: "100%",
+            background: `linear-gradient(to right, #ffffffE6 0%, #ffffffE6 56%, transparent 56%, transparent 100%)`,
+            height: theme.typography.pxToRem(524),
+          }}
+        />
+      </Box>
+      <Section classes={{ root: classes.section }}>
+        <Grid container direction={isMobile ? "column-reverse" : "row"}>
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/components/HowItWorks/index.js:75]: You don't need `isMobile` for this
```suggestion
        <Grid container direction={{ xs: "column-reverse", lg: "row" }}>
```

File: apps/climatemappedafrica/src/components/HowItWorks/index.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    <Box
+      sx={{
+        height: theme.typography.pxToRem(820),
+        position: "relative",
+        [theme.breakpoints.up("md")]: {
+          height: theme.typography.pxToRem(618),
+        },
+        [theme.breakpoints.up("lg")]: {
+          height: theme.typography.pxToRem(728),
+        },
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/components/HowItWorks/index.js:35]: You don't need `theme.breakpoints`:
```suggestion
        height: {  xs: theme.typography.pxToRem(820), md: theme.typography.pxToRem(618), lg: theme.typography.pxToRem(728)},
        position: "relative",
```

or much simpler:
```jsx
        height: {  xs: "820px", md: "618px", lg: "728px" },
        position: "relative",
```

File: apps/climatemappedafrica/src/components/HowItWorks/index.js [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+        sx={{
+          display: {
+            xs: "none",
+            md: "block",
+            lg: "none",
+          },
+        }}
+      >
+        <Box
+          sx={{
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/components/HowItWorks/index.js:64]: Why `Box` in a `Box`? Can't one Box do everything needed?

File: apps/climatemappedafrica/src/components/HowItWorks/index.js [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+            xs={12}
+            md={5}
+            sx={{
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "center",
+              marginBottom: theme.typography.pxToRem(23),
+              [theme.breakpoints.up("md")]: {
+                justifyContent: "flex-end",
+                marginBottom: 0,
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/components/HowItWorks/index.js:164]: Avoid using margins with `Grid` items. `Grid` is implemented using [negative margins](https://mui.com/material-ui/react-grid/#negative-margin) and setting on margins will conflict with that.

File: apps/climatemappedafrica/src/payload/blocks/HowItWorks.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kelvinkipruto) [reply apps/climatemappedafrica/src/payload/blocks/HowItWorks.js]: @kilemensi, It does not change for Youtube videos

[the remark this answers is not in the document]
Comment (kilemensi) [reply apps/climatemappedafrica/src/payload/blocks/HowItWorks.js]: Cool, then lets not ask for it @kelvinkipruto (until we need it).

File: apps/climatemappedafrica/src/components/HowItWorks/index.js [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+                  backgroundColor: "#ffffffE6",
+                  backgroundSize: "120%",
+                },
+                "& .video-js .vjs-big-play-button": {
+                  display: "none",
+                },
+                [theme.breakpoints.up("md")]: {
+                  "& .vjs-poster": {
+                    backgroundSize: "auto",
+                  },
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/components/HowItWorks/index.js:106]: Switch to `backgroundSize: {. xs: ... md: ... }`

File: apps/climatemappedafrica/src/components/HowItWorks/index.js [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+                margin: `${theme.typography.pxToRem(16.5)} 0`,
+                color: theme.palette.grey.dark,
+                width: {
+                  md: theme.typography.pxToRem(278),
+                },
+              })}
+            />
+            <Button href={link.href} variant="text">
+              {link.label}
+            </Button>
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/components/HowItWorks/index.js:143]: Why are you not using a `Link` directly?

File: apps/climatemappedafrica/src/components/HowItWorks/index.js [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+            xs={12}
+            md={5}
+            sx={(theme) => ({
+              display: "flex",
+              alignItems: "center",
+              justifyContent: {
+                xs: "center",
+                md: "flex-end",
+              },
+              marginBottom: theme.typography.pxToRem(23),
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/components/HowItWorks/index.js:157]: I'm still a bit iffy on styling `Grid item` like this; can't all of these styling be done in the `Box` inside the `Grid item`?

File: apps/climatemappedafrica/src/payload/blocks/HowItWorks.js [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        {
+          name: "type",
+          type: "text",
+          required: true,
+          defaultValue: "video/youtube",
+          admin: {
+            description: "The type of video. e.g. video/mp4, video/youtube",
+            hidden: true,
+          },
+        },
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/payload/blocks/HowItWorks.js:53]: If this doesn't change, why are we even asking it (or why not just [hide](https://payloadcms.com/docs/fields/overview#admin-config) it if you want still want the value stored in the CMS)/

File: apps/climatemappedafrica/src/payload/blocks/HowItWorks.js [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+      overrides: {
+        label: {
+          en: "Find out more",
+        },
+        localized: true,
+      },
+    }),
+    image({
+      overrides: {
+        name: "foregroundImage",
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/payload/blocks/HowItWorks.js:67]:
```suggestion
        name: "image",
```

**Review by kilemensi — APPROVED** [review]
LGTM

**Pull request merged**